node/storage.go

```diff
@@ -0,0 +1,184 @@
+package node
+
+import (
+	"fmt"
+
+	"github.com/filecoin-project/storetheindex/store"
+	"github.com/ipfs/go-cid"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+)
+
+var _ store.Storage = &nodeStorage{}
+
+// nodeStorage combines a cache and a persistent storage
+type nodeStorage struct {
+	primary    store.Storage
+	persistent store.PersistentStorage
+}
+
+// NewStorage creates new nodeStorage from a primary and persistent storages
+func NewStorage(primary store.Storage, persistent store.PersistentStorage) *nodeStorage {
+	return &nodeStorage{primary, persistent}
+}
+
+// Get retrieves IndexEntries for a CID
+func (ns *nodeStorage) Get(c cid.Cid) ([]store.IndexEntry, bool, error) {
+	if ns.primary != nil {
+		// Check if CID in primary storage
+		v, found, err := ns.primary.Get(c)
+
```

Comment:
Need to check `err` first.  I do not think we want to hide the error just because the value might be retrievable from persistent storage, because an error at this point means something is broken.
```suggestion
if err != nil {
	return nil, false, err
}
```

Reply:
Good point. My reasoning was that maybe we want the indexer to still be able to give service even if the primary storage breaks, but maybe this is too strong of an assumption. And we can always change it in the future.